source/Calamari.Common/Features/Scripting/ScriptEngine.cs

```diff
@@ -104,12 +108,21 @@ IScriptExecutor GetScriptExecutor(ScriptSyntax scriptSyntax, bool runDotnetScrip
                 case ScriptSyntax.Bash:
                     return new BashScriptExecutor();
                 case ScriptSyntax.FSharp:
+                    LogFSharpDeprecationWarning(variables);
```

Comment:
Can we pass this down into the `FSharpExecutor` class itself so that this class doesn't need to concern itself with this script specific logic?

Reply:
I placed it in the `ScriptEngine` class originally to have the log appear as early as possible (before any of the script wrappers in the chain are executed). But I suppose this doesn't really matter much, and you're right in moving logic down to the `FSharpExecutor` because it should make testing a bit easier I think.